mlflow/data/numpy_dataset.py

```diff
@@ -0,0 +1,156 @@
+                       is automatically computed.
+        """
+        self._features = features
+        self._targets = targets
+        super().__init__(source=source, name=name, digest=digest)
+
+    def _compute_digest(self) -> str:
+        """
+        Computes a digest for the dataset. Called if the user doesn't supply
+        a digest when constructing the dataset.
+        """
+        MAX_ROWS = 10000
+
+        flattened_features = self._features.flatten()
+        trimmed_features = flattened_features[0:MAX_ROWS]
+
+        md5 = hashlib.md5()
+
+        # hash trimmed feature contents
+        try:
+            md5.update(pd.util.hash_array(trimmed_features))
+        except TypeError:
+            md5.update(np.int64(trimmed_features.size))
+        # hash full feature dimensions
+        for x in self._features.shape:
+            md5.update(np.int64(x))
+
+        # hash trimmed targets contents
+        if self._targets is not None:
+            flattened_targets = self._targets.flatten()
+            trimmed_targets = flattened_targets[0:MAX_ROWS]
+            try:
+                md5.update(pd.util.hash_array(trimmed_targets))
+            except TypeError:
+                md5.update(np.int64(trimmed_targets.size))
+            # hash full feature dimensions
+            for x in self._targets.shape:
+                md5.update(np.int64(x))
+
+        return md5.hexdigest()[:8]
+
+    def _to_dict(self, base_dict: Dict[str, str]) -> Dict[str, str]:
+        """
+        :param base_dict: A string dictionary of base information about the
+                          dataset, including: name, digest, source, and source
+                          type.
+        :return: A string dictionary containing the following fields: name,
+                 digest, source, source type, schema (optional), profile
+                 (optional).
+        """
+        base_dict.update(
+            {
+                "schema": json.dumps({"mlflow_tensorspec": self.schema.to_dict()}),
+                "profile": json.dumps(self.profile),
+            }
+        )
+        return base_dict
+
+    @property
+    def source(self) -> FileSystemDatasetSource:
+        return self._source
+
+    @property
+    def features(self) -> Union[np.ndarray, List[np.ndarray], Dict[str, np.ndarray]]:
+        return self._features
+
+    @property
+    def targets(self) -> Union[np.ndarray, List[np.ndarray], Dict[str, np.ndarray]]:
+        return self._targets
+
+    @property
+    def profile(self) -> Optional[Any]:
+        """
+        A profile of the dataset. May be None if no profile is available.
+        """
+        return {
+            "shape": self._features.shape,
+        }
+
+    @cached_property
+    def schema(self) -> Schema:
+        """
+        An MLflow TensorSpec schema representing the tensor dataset
+        """
+        return _infer_schema(self._features)
```

Comment:
Can we try/catch in case schema inference fails and log a warning if it does (see HuggingFace for an example)